### PR TITLE
Fix/transaction query

### PR DESCRIPTION
**Pull request by martiliones**
### Fixed

transforms object from:
```js
const blocks = await api.getTransactions({
  fromHeight: 7585271,
  and: {
    toHeight: 7586280,
  },
  or: {
    senderId: 'U18132012621449491414',
  }
});
```

to 

```html
fromHeight=7585271&and:toHeight=7586280&or:senderId=U18132012621449491414
```

### Added

- Export `transformTransactionQuery` function

### CODE REVIEW

File: src/api/index.ts [hunk 1]
```diff
@@ -99,9 +103,9 @@ export type AddressOrPublicKeyObject = AddressObject | PublicKeyObject;
  */
 export type UsernameOrPublicKeyObject = UsernameObject | PublicKeyObject;
 
-export type TransactionQuery<T extends {}> = T & {
-  or: T;
-  and: T;
+export type TransactionQuery<T extends {}> = Partial<T> & {
```
Comment (bludnic) [inline src/api/index.ts]: Isn't `T extends Record<string, never>` more suitable?

Don't use `{}` as a type. `{}` actually means "any non-nullish value".
- If you want a type meaning "any object", you probably want `object` instead.
- If you want a type meaning "any value", you probably want `unknown` instead.
- **If you want a type meaning "empty object", you probably want `Record<string, never>` instead.**
- If you really want a type meaning "any non-nullish value", you probably want `NonNullable<unknown>` instead.

**Review by bludnic — CHANGES_REQUESTED** [review]

**Comment by adamant-al** [comment]
@martiliones 

Check it.

File: src/api/index.ts [hunk 2]
```diff
@@ -103,7 +103,7 @@ export type AddressOrPublicKeyObject = AddressObject | PublicKeyObject;
  */
 export type UsernameOrPublicKeyObject = UsernameObject | PublicKeyObject;
 
-export type TransactionQuery<T extends {}> = Partial<T> & {
+export type TransactionQuery<T extends object> = Partial<T> & {
```
Comment (bludnic) [inline src/api/index.ts:106]: Why you use `object` instead of `Record<string, never>`?

**Review by bludnic — APPROVED** [review]

File: src/api/index.ts (hunk 2 above)
Comment (martiliones) [reply src/api/index.ts:106]: because we need an object with existing properties

**Pull request merged**